ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -80,58 +91,119 @@ internal fun StackComponentView(
     )
 
     if (stackState.visible) {
-        val backgroundColorStyle = stackState.backgroundColor?.let { rememberColorStyle(scheme = it) }
-        val borderStyle = stackState.border?.let { rememberBorderStyle(border = it) }
-        val shadowStyle = stackState.shadow?.let { rememberShadowStyle(shadow = it) }
+        if (style.badge != null) {
+            when (style.badge.style) {
+                Badge.Style.Overlay -> {
+                    StackWithOverlaidBadge(stackState, state, style.badge.stackStyle, style.badge.alignment, modifier)
+                }
 
-        // Modifier irrespective of dimension.
-        val commonModifier = remember(stackState, backgroundColorStyle, borderStyle, shadowStyle) {
-            Modifier
-                .padding(stackState.margin)
-                .applyIfNotNull(shadowStyle) { shadow(it, stackState.shape) }
-                .applyIfNotNull(backgroundColorStyle) { background(it, stackState.shape) }
-                .clip(stackState.shape)
-                .applyIfNotNull(borderStyle) { border(it, stackState.shape) }
-                .padding(stackState.padding)
-        }
+                Badge.Style.EdgeToEdge -> {
+                }
 
-        val content: @Composable () -> Unit = remember(stackState.children) {
-            @Composable { stackState.children.forEach { child -> ComponentView(style = child, state = state) } }
+                Badge.Style.Nested -> {
+                }
+            }
+        } else {
+            MainStackComponent(stackState, state, modifier)
         }
+    }
+}
 
-        // Show the right container composable depending on the dimension.
-        when (val dimension = stackState.dimension) {
-            is Dimension.Horizontal -> Row(
-                modifier = modifier
-                    .size(stackState.size, verticalAlignment = dimension.alignment.toAlignment())
-                    .then(commonModifier),
-                verticalAlignment = dimension.alignment.toAlignment(),
-                horizontalArrangement = dimension.distribution.toHorizontalArrangement(
-                    spacing = stackState.spacing,
-                ),
-            ) { content() }
+@Composable
+private fun StackWithOverlaidBadge(
+    stackState: StackComponentState,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
```

Comment:
Another heads up: I _think_ this composable should turn the `badgeStack` into a `StackComponentState` using `rememberUpdatedStackComponentState()`, just like we do with the main `StackComponentStyle`. Can probably happen internally? Not sure, haven't thought long about it haha.